src/OrleansAzureUtils/Providers/Storage/AzureTableStorage.cs

```diff
@@ -234,9 +237,10 @@ internal void ConvertToStorageFormat(object grainState, GrainStateEntity entity)
 
                 if (Log.IsVerbose3) Log.Verbose3("Writing JSON data size = {0} for grain id = Partition={1} / Row={2}",
                     data.Length, entity.PartitionKey, entity.RowKey);
-                
-                dataSize = data.Length;
-                entity.StringData = data;
+
+                CheckMaxDataSize(data.Length, MAX_DATA_CHUNK_SIZE);
+
+                entity.Properties.Add(STRING_DATA_PROPERTY_NAME, new EntityProperty(data));
```

Comment:
> only catch is that data size is string.lenth*2 because of Unicode.
It seems that there was actually a bug before, since the string length was compared against 64*1024.